### PR TITLE
spec for write protection error

### PR DESCRIPTION
**Pull request by DreamWuGit**
- add hackmd for write protection error
- TODO: add python code after circuit pr (https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1001) merged.

### CODE REVIEW

File: specs/error_state/error_write_protection.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# ErrorWriteProtection state
+
+## Procedure
+there are some op codes which modify state. there are `[SSTORE, CREATE, CREATE2, 
+CALL, SELFDESTRUCT, LOG0, LOG1, LOG2, LOG3, LOG4]`
+when execution call context is ready only (static call), these op codes running will encounter
+write protection error.
+### EVM behavior
+in above op codes which modify state, `CALL` is somewhat special. for non call op codes(SSTORE, CREATE, etc)
+first check if running in ready only call context, if yes, throw write protection error.
```
Comment (lispc) [inline specs/error_state/error_write_protection.md:10]: if running in ready only call context => if running in 【read？】 only call context

File: specs/error_state/error_write_protection.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+for `CALL` op code, it will also check `value` is not zero. only both non zero value & ready only call context, then
```
Comment (lispc) [inline specs/error_state/error_write_protection.md:11]: ready only => read only

File: specs/error_state/error_write_protection.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+throw write protection error.
+
+### Constraints
+1. constrain this error happens in one of op codes `[SSTORE, CREATE, CREATE2, 
+CALL, SELFDESTRUCT, LOG0, LOG1, LOG2, LOG3, LOG4]`
+2. current call context must be readonly & internal call(since requires at least one `staticcall` earlier).
```
Comment (lispc) [inline specs/error_state/error_write_protection.md:17]: i am not sure whether "& internal call" is needed.   "call context be readonly" must be inside a staticcall?

File: specs/error_state/error_write_protection.md (hunk 1 above)
Comment (DreamWuGit) [reply specs/error_state/error_write_protection.md]: fix typo in [7233568](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/365/commits/7233568918063a0170ea894d7e7219e9e822fce6) thanks for catching.

File: specs/error_state/error_write_protection.md (hunk 2 above)
Comment (DreamWuGit) [reply specs/error_state/error_write_protection.md]: fixed in [7233568](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/365/commits/7233568918063a0170ea894d7e7219e9e822fce6) thanks.

File: specs/error_state/error_write_protection.md (hunk 3 above)
Comment (DreamWuGit) [reply specs/error_state/error_write_protection.md:17]: describe it here to make it more clear.

**Comment by DreamWuGit** [comment]
> @DreamWuGit it's ok to you to address current comments in order to finish circuit PR?

@adria0 done.

**Comment by ed255** [comment]
> In order to simplify I think that is better to create a new PR with the python specification and closing this one. This two-step md-spec > python-spec approvals are quite messy.

What about extending this PR with the python spec, now that the rust implementation is already done?  @DreamWuGit could you do this?

**Comment by DreamWuGit** [comment]
> What about extending this PR with the python spec, now that the rust implementation is already done? @DreamWuGit could you do this?

working on other objects now, will merge this one and keep #357 open until late python implementation done !

**Pull request merged**